### PR TITLE
Fix 1376 turrets damage can decrease

### PR DESCRIPTION
**Pull request by vmarkovtsev**
Try to fix http://bugs.vcmi.eu/view.php?id=1376

I've chosen way #3 (hardcode) as the most simple one.

### CODE REVIEW

File: lib/CBattleCallback.cpp [hunk 1]
```diff
@@ -879,37 +885,50 @@ TDmgRange CBattleInfoCallback::calculateDmgRange(const BattleAttackInfo &info) c
...
+	{
+		double multAttackReduction =
+			(100 - battleBonusValue(info.attackerBonuses,
+			                        Selector::type(Bonus::GENERAL_ATTACK_REDUCTION)))
+			/ 100.0;
+		attackDefenceDifference +=
+			battleBonusValue(info.attackerBonuses,
+			                 Selector::typeSubtype(Bonus::PRIMARY_SKILL,
+			                                       PrimarySkill::ATTACK))
+			* multAttackReduction;
```
Comment (DjWarmonger) [inline lib/CBattleCallback.cpp]: Here we can see that 80-char limit leads to multi-line zig-zags :/

Comment (vmarkovtsev) [inline lib/CBattleCallback.cpp]: Matter of habit. All PEP8-conformant (the majority) Python code looks the same. Actually, my editor font is 12-14 (I see bad) and on 13" macbook it was the only way to view everything (no splitscreens).

Comment (DjWarmonger) [inline lib/CBattleCallback.cpp]: At least aligne them to left ot right so we could see it's one piece of code. Now every line has different alignment :v

**Comment by ArseniyShestakov** [comment]
Hardcoding is reasonable, but not in case when we already have appropriate way to do things. AVS suggested to solve this by adding ENEMY_DEFENCE_REDUCTION bonus to arrow towers (#136). I there any reason this can't be done that way?

File: lib/CBattleCallback.cpp (hunk 1 above)
Comment (vmarkovtsev) [inline lib/CBattleCallback.cpp]: It is standard to format code this way (align parameters by function boundary), but I don't insist - if you suggest another formatting style to fit into 80 - OK. Could you post an example?

**Comment by DjWarmonger** [comment]
ENEMY_DEFENCE_REDUCTION set to 100 means that effective creature defense will be always zero, but turret attack can go high and result in damage bonus. Is that desired?

**Comment by ArseniyShestakov** [comment]
As far as I know that exactly what we want in case of arrow towers.

**Comment by vmarkovtsev** [comment]
I'll see what I can do with this bonus

**Comment by vmarkovtsev** [comment]
Wow, that was easy :)

File: config/creatures/special.json [hunk 2]
```diff
@@ -112,7 +112,11 @@
 		"index": 149,
 		"level": 0,
 		"faction": "neutral",
-		"abilities": { "shooter" : { "type" : "SHOOTER" } },
+		"abilities":
+		{
+			"shooter" : { "type" : "SHOOTER" },
+			"reduceDefence" : { "type" : "ENEMY_DEFENCE_REDUCTION", "val" : 100 }
```
Comment (alexvins) [inline config/creatures/special.json]: "ingoreDefence" is better name for this ability

**Comment by DjWarmonger** [comment]
We've desined awesome mod system for a purpose ;)

**Pull request merged**